hack/run_test

```diff
@@ -350,6 +352,43 @@ function setup_replication_cluster() {
   test_slave_visibility $cid_suffix
 }
 
+function run_master_restart_test() {
+  echo "Testing failed master restart"
+  local cluster_args="-e POSTGRESQL_ADMIN_PASSWORD=pass -e POSTGRESQL_MASTER_USER=$PGUSER -e POSTGRESQL_MASTER_PASSWORD=$PASS"
+  local cid_suffix="mrestart"
+  local table_name="t1"
+
+  local volume_dir
+  volume_dir=`mktemp -d --tmpdir pg-testdata.XXXXX`
+  chmod a+rwx ${volume_dir}
+  master_args="-v ${volume_dir}:/var/lib/pgsql/data:Z"
+
+  # Setup the cluster
+  local slave_num=2
+  setup_replication_cluster $slave_num
```

Comment:
setup_replication_cluster takes the local $slave_num variable as an optional argument and sets up a local variable with the same name inside the function, defaulting to 1 if called without the argument.
As both of the variables are local, there should not be any value transfer by variable going on.